### PR TITLE
Java console Daemon sample

### PR DESCRIPTION
**Pull request by ramya25**
Java console client credential grant sample (with client secret)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,64 +1,201 @@
...
+Open the `ClientCredentialGrant.java` to configure the project.
+
+#### Configure the client project
+
+1. Open the `src\main\java\ClientCredentialGrant` class
+1. Find the line `final static String TENANT_SPECIFIC_AUTHORITY` and replace `Enter_the_Tenant_Info_Here` with your Azure AD **Tenant Id**.
+1. Find the line `final static String CONFIDENTIAL_CLIENT_ID` and replace the existing value with the **Application ID (clientId)** of the `java-daemon-console` application copied from the Azure portal.
+1. Find the line `final static String CONFIDENTIAL_CLIENT_SECRET` and replace the existing value with the **key value** you saved during the creation of the `daemon-console` app, in the Azure portal.
+
+### Step 4: Run the sample
```
Comment (navyasric) [inline README.md]: I think we need to add the steps to run the Java sample similar to other samples

File: src/main/java/ClientCredentialGrant.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    private static String getUsersListFromGraph(String accessToken) throws IOException {
+        URL url = new URL("https://graph.microsoft.com/v1.0/users");
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+
+        conn.setRequestMethod("GET");
+        conn.setRequestProperty("Authorization", "Bearer " + accessToken);
+        conn.setRequestProperty("Accept","application/json");
+
+        int httpResponseCode = conn.getResponseCode();
+        if(httpResponseCode == HTTPResponse.SC_OK) {
```
Comment (sangonzal) [inline src/main/java/ClientCredentialGrant.java]: Here we can do httpResponseCode == 200, and that way we won't need oauth2-oidc-sdk

File: README.md [hunk 3]
```diff
@@ -1,64 +1,201 @@
...
+languages:
+  - java
+products:
+  - azure
+  - azure-active-directory
+  - office-ms-graph
+description: "Shows how a daemon console app uses MSAL4J to get an access token and call Microsoft Graph."
+---
+
+# A Java simple daemon console application calling Microsoft Graph with its own identity
```
[the remark this answers is not in the document]
Comment (ramya25) [reply README.md]: updated

File: README.md [hunk 4]
```diff
@@ -1,64 +1,201 @@
...
+
+### Step 1:  Clone or download this repository
+
+From your shell or command line:
+
+```Shell
+git clone https://github.com/Azure-Samples/ms-identity-java-daemon.git
+```
+
+or download and exact the repository .zip file.
```
[the remark this answers is not in the document]
Comment (ramya25) [reply README.md]: Thanks, corrected.

File: README.md (hunk 1 above)
Comment (ramya25) [reply README.md]: Added.

File: README.md [hunk 5]
```diff
@@ -1,64 +1,217 @@
...
+
+![Topology](./ReadmeFiles/topology.png)
+
+For more information on the concepts used in this sample, be sure to read the [Microsoft identity platform endpoint client credentials protocol documentation](https://azure.microsoft.com/documentation/articles/active-directory-v2-protocols-oauth-client-creds).
+
+## How to run this sample
+
+To run this sample, you'll need:
+
+- Working installation of [Java](http://www.oracle.com/technetwork/java/javase/downloads/index.html) and [Maven](https://maven.apache.org/).
```
[the remark this answers is not in the document]
Comment (ramya25) [reply README.md]: Updated.

File: src/main/java/ClientCredentialGrant.java (hunk 2 above)
Comment (ramya25) [reply src/main/java/ClientCredentialGrant.java]: updated.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    <artifactId>msal-Client-Credential-Grant-Sample</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.microsoft.azure</groupId>
+            <artifactId>msal4j</artifactId>
+            <version>1.0.0</version>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (ramya25) [reply pom.xml]: Yes makes sense, hardcoded to 200 (httpResponseCode == 200) and removed oauth2-oidc-sdk, also org.json.json is not being used so removed. 
Sample runs fine but I just see an error saying `Failed to load class "org.slf4j.impl.StaticLoggerBinder"`, if I dont add org.slf4j.slf4j-simple in the dependencies.

**Pull request merged**